# DP: Avoid linker error on 1480 because sb1250_hpt_setup() is not defined
# DP: Upstream status: in 2.6.18-rc1

Date:	Thu, 16 Mar 2006 14:11:27 +0000
From: Martin Michlmayr <tbm@cyrius.com>
To: linux-mips@linux-mips.org
Subject: Re: [MIPS] Sibyte: Fix race in sb1250_gettimeoffset().

* linux-mips@linux-mips.org <linux-mips@linux-mips.org> [2006-03-16 12:57]:
> Commit: 186326fa1e0360450b927ee5b21fb8db028fe7ba
> 
> +void __init swarm_time_init(void)
> +{
> +	/* Setup HPT */
> +	sb1250_hpt_setup();
> +}

This leads to compiler errors on 1480 because sb1250_hpt_setup() is
not defined.  We need something like the patch below (or possibly a
proper fix?):


[MIPS] don't call sb1250_hpt_setup on 1480

sb1250_hpt_setup() should not be called on the 1480 board since it's
note defined there, leading to a linking error.

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>


diff --git a/arch/mips/sibyte/swarm/setup.c b/arch/mips/sibyte/swarm/setup.c
index b661d24..4a93f1d 100644
--- a/arch/mips/sibyte/swarm/setup.c
+++ b/arch/mips/sibyte/swarm/setup.c
@@ -72,8 +72,10 @@ const char *get_system_type(void)
 
 void __init swarm_time_init(void)
 {
+#if defined(CONFIG_SIBYTE_SB1250) || defined(CONFIG_SIBYTE_BCM112X)
 	/* Setup HPT */
 	sb1250_hpt_setup();
+#endif
 }
 
 void __init swarm_timer_setup(struct irqaction *irq)

